cppcache/src/TcrConnection.cpp

```diff
@@ -659,16 +656,15 @@ char* TcrConnection::readMessage(size_t* recvLen,
       if (isNotificationMessage) {
         // fix #752 - do not throw periodic TimeoutException for subscription
         // channels to avoid frequent stack trace processing.
-        return nullptr;
+        return {nullptr, 0, error};
```

Comment:
I love tuple!

Reply:
I do to as a concept, use it all the time in my Python code.  Just not a fan of the C++ syntax, like I said I find it clumsy.  Of course, that's true of a lot of the language :).

Reply:
C++17 makes it all good. 
https://en.cppreference.com/w/cpp/language/structured_binding